ambry-store/src/main/java/com.github.ambry.store/BlobStore.java

```diff
@@ -235,6 +236,15 @@ public StoreInfo get(List<? extends StoreKey> ids, EnumSet<StoreGetOptions> stor
         BlobReadOptions readInfo = index.getBlobReadInfo(key, storeGetOptions);
         readOptions.add(readInfo);
         indexMessages.put(key, readInfo.getMessageInfo());
+        // validate accountId and containerId
+        if (config.storeGetAuthorizationCheck && !Utils.validateAuthorization(
```

Comment:
I think that we should always do the auth check but throw exception only if config is true, otherwise fire a metric. This way we can "test-run" this, make sure we haven't missed any corner cases and then turn on the config.

Something like
```
if (!authCheckOk()) {
  if (config) {
    // throw exception
  } else {
    // fire metric
  }
}
```

Reply:
Sounds good. I will use "metrics.getAuthorizationFailure.inc();", which is also used by AmbryRequest.